packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -249,393 +314,657 @@ describe("Decentraland", () => {
+
+              done();
+            };
+
+            const errorListener = error => {
+              action.off("error");
+              done(error);
+            };
+
+            action.once("confirmation", confirmationListener);
+            action.on("error", errorListener);
+
+            action.send();
+          })();
         });
       });
 
       // WIP: Not working because of gas issue on Marketplace.safeExecuteOrder() call
-      describe.skip("Using funds from a Gnosis Safe wallet", () => {
-        beforeEach("onboarding", async () => {
-          // Funding ephemeral account with some ethers to pay for gas
-          // TODO: ephemeralWallet should broadcast this action
-          const toAddress = ephemeralAddress;
+      describe("Using funds from a Gnosis Safe wallet", () => {
+        beforeEach("onboarding - funding ephemeral account with ETH", done => {
           gnosisSafe.setSigners([gnosisSafeOwner]);
           gnosisSafe.setWallet(gnosisSafeOwner);
-          const transferEthersAction = gnosisSafe.transferEther(
-            toAddress,
-            SMALL_AMOUNT
-          );
-          await transferEthersAction.send();
-          await transferEthersAction.waitForOneConfirmation();
-          await expectExactEtherBalances(provider, [toAddress], [SMALL_AMOUNT]);
-
-          // Gnosis Safe should approve Marketplace to spend its MANA Tokens
-          // TODO: ephemeralWallet should broadcast this action
-          const contractAddress = mana.getAddress();
-          const contractABI = mana.getABI();
-          const functionName = "approve";
-          const argsArray = [MARKETPLACE_ADDRESS, manaAmountForShopping];
-          gnosisSafe.setSigners([gnosisSafeOwner]);
-          gnosisSafe.setWallet(gnosisSafeOwner);
-          const approvalAction = gnosisSafe.customContractAction(
-            contractAddress,
-            contractABI,
-            functionName,
-            argsArray
-          );
-          await approvalAction.send();
-          await approvalAction.waitForOneConfirmation();
-
-          const owner = GNOSIS_SAFE_ADDRESS;
-          const spender = MARKETPLACE_ADDRESS;
-          const allowance = await mana.allowance(owner, spender);
-          expect(`${allowance}`).to.equal(`${manaAmountForShopping}`);
-        });
-
-        it("should buy an estate", async () => {
-          const {
-            assetId,
-            nftAddress,
-            seller,
-            priceInWei,
-            expiresAt,
-          } = estateForSale;
-
-          const buyerAddress = GNOSIS_SAFE_ADDRESS;
-          await checkAsset(estate, mana, estateForSale, buyerAddress);
-
-          await expectExactTokenBalances(estate, [GNOSIS_SAFE_ADDRESS], [0]);
+          const toAddress = ephemeralAddress;
 
-          // Gnosis Safe should execute an open order
+          // Funding ephemeral account with some ethers to pay for gas
           // TODO: ephemeralWallet should broadcast this action
```

Comment:
That's an interesting idea - curious for your rationale here

Reply:
Then it's more like a Tasit app clicking a "fund myself" button?

Reply:
Yes.

Reply:
I guess if a user in the Tasit app clicked fund myself, it would still work just as well if it deep-linked into Gnosis Safe app and then the Gnosis Safe app broadcast it, right?

Reply:
I think so, that test is using the gnosisOwner to call the gnosisSafe contract. So it's like to **not** having the `ephemeral` account as a signer and calling the gnosisSafeApp thru deep-link.

Reply:
Yep. So depending on the scope of this particular test, either option is a sensible choice.

Maybe add something from this GitHub comment thread to explain that in addition to the current TODO comment in the code?

Reply:
Ok, I'll do that.